smart-contracts/test/Lock/erc721/approve.js

```diff
@@ -41,53 +41,55 @@ contract('Lock ERC721', (accounts) => {
       })
 
       describe('when the sender is not the token owner', () => {
-        it('should fail', () => {
-          return locks['FIRST']
-            .approve(accounts[2], accounts[1], {
-              from: accounts[2]
-            })
-            .catch(error => {
-              assert.equal(error.message, 'VM Exception while processing transaction: revert')
-            })
+        it('should fail', async () => {
+          try {
+            let ID = await locks['FIRST'].getTokenIdFor(accounts[1])
+            await locks['FIRST']
+              .approve(accounts[2], ID, {
+                from: accounts[2]
+              })
+          } catch (error) {
+            assert.equal(error.message, 'VM Exception while processing transaction: revert')
```

Comment:
Hmm, why is there no revert reason in this error?

Reply:
Fixed. Added reason to contract and updated test cases (x2)